mapper-pojo/src/main/java/org/hibernate/search/v6poc/entity/pojo/bridge/builtin/spatial/impl/GeoPointBridgeImpl.java

```diff
@@ -6,45 +6,46 @@
  */
 package org.hibernate.search.v6poc.entity.pojo.bridge.builtin.spatial.impl;
 
+import java.util.Objects;
 import java.util.function.Function;
 import java.util.stream.Collector;
 
+import org.hibernate.search.v6poc.backend.document.model.Store;
 import org.hibernate.search.v6poc.backend.document.model.spi.IndexSchemaElement;
 import org.hibernate.search.v6poc.backend.document.spi.DocumentState;
 import org.hibernate.search.v6poc.backend.document.spi.IndexFieldAccessor;
 import org.hibernate.search.v6poc.backend.spatial.GeoPoint;
-import org.hibernate.search.v6poc.entity.model.spi.SearchModel;
-import org.hibernate.search.v6poc.entity.pojo.bridge.builtin.spatial.GeoPointBridge;
 import org.hibernate.search.v6poc.backend.spatial.ImmutableGeoPoint;
+import org.hibernate.search.v6poc.entity.model.spi.SearchModel;
 import org.hibernate.search.v6poc.entity.pojo.bridge.spi.Bridge;
-import org.hibernate.search.v6poc.engine.spi.BuildContext;
-import org.hibernate.search.v6poc.entity.pojo.model.spi.PojoState;
-import org.hibernate.search.v6poc.entity.pojo.model.spi.PojoModelElementAccessor;
 import org.hibernate.search.v6poc.entity.pojo.model.spi.PojoModelElement;
+import org.hibernate.search.v6poc.entity.pojo.model.spi.PojoModelElementAccessor;
+import org.hibernate.search.v6poc.entity.pojo.model.spi.PojoState;
 import org.hibernate.search.v6poc.util.SearchException;
 import org.hibernate.search.v6poc.util.StreamHelper;
 
 /**
  * @author Yoann Rodiere
  */
-public class GeoPointBridgeImpl implements Bridge<GeoPointBridge> {
+public class GeoPointBridgeImpl implements Bridge {
 
-	private GeoPointBridge parameters;
+	private final String fieldName;
+	private final Store store;
+	private final String markerSet;
```

Comment:
My only comment is that I'm a bit worried that we have to declare these parameters several times. It feels a bit tedious to me.

Reply:
I understand your concern, but that's not something we can avoid. With this approach, it's actually the annotations that duplicate the parameters that are originally declared in the builder. And obviously annotations must be statically defined, so they need to re-declare the parameters.
Either we think it's acceptable, and then we can merge the PR, or we don't, and then I will have to give up about this.

Some remarks to reassure you:

* Even with the previous solution (the one used in Validator, i.e. a "definition" that builds an annotation instance dynamically), we already need to declare the parameters several times: once in the annotation and once in the "definition" setters. Here we simply have a third declaration for storing the values inside the builder, but hey, that's how bean properties work in Java :)
* If it's declaring "standard" parameters ("store", "fieldName", "analyzer", ...) that worries you, we could simply provide an abstract class implementing all of those, and let people extend it.
* Finally, let's keep in mind that writing a custom bridge, especially one with many parameters, is not something we expect users to do constantly. It's more of a "write once, use everywhere" situation, so a little bit of verbosity when writing the bridge is not the end of the world.

Reply:
The difference with HV is that the `YourConstraintDef` class should be added only if you want to use the programmatic API. You can define a constraint and a validator without defining this additional class.

IMHO, there are 2 different things in my comment:
 - yeah the fact that we need to redeclare the common attributes every time is annoying. I'm not sure we can do better unfortunately, considering the limitations of annotations (you can have a common parent for the API part but not for the annotation unfortunately). Something like `@FieldBridge(bridge = @MyBridge(...), commonParameter1 = ...)` would be ideal. We had the same issue with HV :/. Not sure what we can do to mitigate the issue.
- the quantity of boilerplate code needed to create a bridge. I agree it's not something you will do very often. But I'm wondering if the step will not be too high and if we will have users give up on this.

Agreed that it only concerns advanced bridges.

No real solution here, just wanted to share my concerns from a user POV.